### ISSUE TITLE
Setting the number of labels used for semi-supervising

### ISSUE THREAD

**Issue by daiisuke27** [tjucwb/CatGAN#1]
Hi,
I have an issue about the setting of the number of labels used in the training, let's say that i want for a semi supervised learning to used the entire MNIST set but only with 100 labelled samples, how should i edit the code to get what i want ?
Thanks.

**Comment by tjucwb** [comment]
Just add a extra term in the diacriminator loss, the cross entropy between the predicted label and the true label as the paper shows in equation 9 (torch.CrossEntropyLoss(gnet(img),label))

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by tjucwb** [comment]
Just add a extra term in the discriminator loss, the cross entropy between the predicted label and the true label as the paper shows in equation 9  (torch.CrossEntropyLoss(d_net(r_img),label))

**Comment by daiisuke27** [comment]
I have tried to do what you said but failed. So could you answer my following questions :
1- Are you using a semi supervised learning as in the paper? because when i check the code i see that you have loaded all training samples with their labels, but in the paper they only used 10 or 100 labelled samples and used the rest of the samples without labels.
2- If i want to, for example : use only 100 labelled samples and use the other samples without labels, what should i edit in your code ?
PS : i know that you have answered my question but i am just starting with GAN's, so i just hope that you give me a more detailed answer
Thank you so much !

**Comment by tjucwb** [comment]
1. I loaded them but did not use them, so it ia unsupervised
2. For semi supervision, u can just use 10/100labelled data,and tge labelled data should be fuxed. Generally the images uswd for training are different in every iteration, but for semi aupervision, u can only use same 100 labelled data in each iteration the CrossEntropyLoss of the 100 labelled data should be added in the diacriminator loss for training in every iteration.

Hope it is clear
Good luck
________________________________

**Comment by tjucwb** [comment]
I uploaded a example of semi-supervision of CatGAN in main2.py (not so beautiful, need to be improved, so it is in another file), I just show you a example according to my understanding to the paper, unfortunately the accuracy is jut 96% with the current setting, if you want to get a better accuracy, u need to tune the parameters by yourself, cauz this script is just a demo, that shows the idea works more or less.
u can run the file with 
python main2.py --semi

**Comment by daiisuke27** [comment]
Thank you so much ! your code helped me a lot to understand the paper too.
Greetings.

**Comment by makelei-2019** [comment]
Hi
where i can find the paper?
thanks

**Comment by tjucwb** [comment]
just google catgan

**Comment by makelei-2019** [comment]
oh  谢谢
